torchvision/transforms/functional.py

```diff
@@ -82,6 +82,38 @@ def to_tensor(pic):
         return img
 
 
+def convert_image_dtype(
+    image: torch.Tensor, dtype: torch.dtype = torch.float
+) -> torch.Tensor:
+    def scale_factor(dtype: torch.dtype) -> float:
+        if dtype in (
+            torch.float32,
+            torch.float,
+            torch.float64,
+            torch.double,
+            torch.float16,
+            torch.half,
+        ):
+            return 1.0
+
+        num_value_bits = {
+            torch.uint8: 8,
+            torch.int8: 7,
+            torch.int16: 15,
+            torch.short: 15,
+            torch.int32: 31,
+            torch.int: 31,
+            torch.int64: 63,
+            torch.long: 63,
+            torch.bool: 1,
+        }
+        return float(2 ** num_value_bits[dtype] - 1)
+
+    return (
+        image.double().div(scale_factor(image.dtype)).mul(scale_factor(dtype)).to(dtype)
```

Comment:
Why the cast to double? Also, it would be more readable to use `+-*/` operators instead of `mul` and `div`

Reply:
Also, I think the `saturate` option is good to have, can you add it?

Reply:
> Why the cast to double?

I think we should cast to floating point first to avoid rounding errors:

```python
import torch

image = torch.tensor(254, dtype=torch.uint8)
image / 255 == torch.tensor(0, dtype=torch.uint8)  # True
```
Of all floating point types `double` is simply the largest and needed if `image.dtype == torch.double`.

> Also, it would be more readable to use +-*/ operators instead of mul and div

Sure.

Reply:
Note that we now perform automatic type promotion depending on the operands, so
```python
image = torch.tensor(254, dtype=torch.uint8)
image / 255.0 == tensor(0.9961)
```

so we could avoid converting to double and then to float (making the operation quite a bit more costly)

Reply:
Is this new in `torch>1.4.0`? At least locally this does not work with `torch==1.4.0`. If this works from now on I agree to cast to double is obsolete.